tests/eigensolve_test.cpp

```diff
@@ -75,303 +58,86 @@ void display_test_info()
   add_eigen_option_group(app);
-  // add_deflation_option_group(app);
-  // add_multigrid_option_group(app);
   try {
     app->parse(argc, argv);
   } catch (const CLI::ParseError &e) {
     return app->exit(e);
   }
 
-  if (prec_sloppy == QUDA_INVALID_PRECISION) prec_sloppy = prec;
-  if (link_recon_sloppy == QUDA_RECONSTRUCT_INVALID) link_recon_sloppy = link_recon;
-  if (link_recon_precondition == QUDA_RECONSTRUCT_INVALID) link_recon_precondition = link_recon_sloppy;
+  // Set some default values for precisions if none are passed through the command line
+  setQudaDefaultPrecs();
 
-  // initialize QMP/MPI, QUDA comms grid and RNG (test_util.cpp)
+  // initialize QMP/MPI, QUDA comms grid and RNG (host_utils.cpp)
   initComms(argc, argv, gridsize_from_cmdline);
 
-  // call srand() with a rank-dependent seed
-  initRand();
+  // Only these fermions are supported in this file
+  if (dslash_type != QUDA_WILSON_DSLASH && dslash_type != QUDA_CLOVER_WILSON_DSLASH
+      && dslash_type != QUDA_TWISTED_MASS_DSLASH && dslash_type != QUDA_DOMAIN_WALL_4D_DSLASH
+      && dslash_type != QUDA_MOBIUS_DWF_DSLASH && dslash_type != QUDA_TWISTED_CLOVER_DSLASH
+      && dslash_type != QUDA_DOMAIN_WALL_DSLASH) {
+    printfQuda("dslash_type %d not supported\n", dslash_type);
+    exit(0);
+  }
 
-  // QUDA parameters begin here.
-  //------------------------------------------------------------------------------
   QudaGaugeParam gauge_param = newQudaGaugeParam();
-  setGaugeParam(gauge_param);
+  setWilsonGaugeParam(gauge_param);
 
-  QudaEigParam eig_param = newQudaEigParam();
   // Though no inversions are performed, the inv_param
   // structure contains all the information we need to
   // construct the dirac operator. We encapsualte the
   // inv_param structure inside the eig_param structure
   // to avoid any confusion
   QudaInvertParam eig_inv_param = newQudaInvertParam();
   setInvertParam(eig_inv_param);
+  // Specific changes to the invert param for the eigensolver
+  eig_inv_param.gamma_basis = QUDA_UKQCD_GAMMA_BASIS;
+  eig_inv_param.solve_type
+    = (eig_inv_param.solution_type == QUDA_MAT_SOLUTION ? QUDA_DIRECT_SOLVE : QUDA_DIRECT_PC_SOLVE);
+  QudaEigParam eig_param = newQudaEigParam();
+  // Place Invert param inside Eig param
   eig_param.invert_param = &eig_inv_param;
   setEigParam(eig_param);
 
-  // All user inputs now defined
+  // Initialize the QUDA library
+  initQuda(device);
   display_test_info();
 
-  // set parameters for the reference Dslash, and prepare fields to be loaded
+  // Set some dimension parameters
   if (dslash_type == QUDA_DOMAIN_WALL_DSLASH || dslash_type == QUDA_DOMAIN_WALL_4D_DSLASH
       || dslash_type == QUDA_MOBIUS_DWF_DSLASH) {
     dw_setDims(gauge_param.X, eig_inv_param.Ls);
   } else {
     setDims(gauge_param.X);
   }
-
-  // set spinor site size
-  int sss = 24;
-  if (dslash_type == QUDA_LAPLACE_DSLASH) sss = 6;
+  // Set spinor site size
+  int sss = (dslash_type == QUDA_LAPLACE_DSLASH ? 6 : 24);
```

Comment:
looks like LAPLACE is not listed as compatible above

Reply:
Yep, laplace is in the staggered file only, fixed with 1fdd1152c